src/Process/Runner/MutationTestingRunner.php

```diff
@@ -82,6 +82,7 @@ public function run(iterable $mutations, string $testFrameworkExtraOptions): voi
         $this->eventDispatcher->dispatch(new MutationTestingWasStarted($numberOfMutants, $this->processRunner));
 
         $processContainers = take($mutations)
+            ->stream()
```

Comment:
> Ensures subsequent operations use lazy, non-array paths

could you please add more explanation about what it changes exactly, I'm not sure I understand.

https://github.com/sanmai/pipeline/blob/a293c031a999e0fa954507ff9c06a68aa13974c1/src/Standard.php#L665-L683

it just forces it to be Generator? And how does it changes anything?

Without `--no-progress` (by default) we already wait on line 81:

```
        $numberOfMutants = IterableCounter::bufferAndCountIfNeeded($mutations, $this->runConcurrently);
```

here, no?

Reply:
The hold-up happens after this line, when we actually start processing the mutations as in: run tests and see what tests fail.

So before we had one mode that was fully array-based (non-concurrent), and we had a lazy concurrent generator-based mode. Now we have a full generator-based mode, an a partial generator-based mode that collects mutations at one point to count them, and then resumes the generator-based approach. What does not make sense here?

Reply:
https://github.com/user-attachments/assets/6d336bf7-5b5b-4d02-8a3e-110696c2a05f

Let's review. I'm uploading it now, editing the comment later.